compiler/rustc_codegen_llvm/src/gotoc/place.rs

```diff
@@ -208,26 +208,26 @@ impl<'tcx> GotocCtx<'tcx> {
     /// code for Fn::call and Fn::call_mut both use pointers to a FnDef.
     /// In these cases, we need to generate an expression that references the
     /// existing fndef rather than a named variable.
-    pub fn codegen_local_fndef(&mut self, l: Local) -> Option<Expr> {
-        let t = self.local_ty(l);
-        match t.kind() {
+    pub fn codegen_local_fndef(&mut self, ty: &'tcx ty::TyS<'tcx>) -> Option<Expr> {
+        match ty.kind() {
             // A local that is itself a FnDef, like Fn::call_once
             ty::FnDef(defid, substs) => Some(self.codegen_fndef(*defid, substs, None)),
-            // A local that is a pointer to a FnDef, like Fn::call and Fn::call_mut
-            ty::RawPtr(inner) => match inner.ty.kind() {
-                ty::FnDef(defid, substs) => {
-                    Some(self.codegen_fndef(*defid, substs, None).address_of())
-                }
-                _ => None,
-            },
+            // A local can be pointer to a FnDef, like Fn::call and Fn::call_mut
+            ty::RawPtr(inner) => self.codegen_local_fndef(inner.ty).map(|f| f.address_of()),
+            // A local can be a boxed function pointer
+            ty::Adt(def, _) if def.is_box() => {
+                let boxed_ty = self.codegen_ty(ty);
+                self.codegen_local_fndef(ty.boxed_ty())
+                    .map(|f| self.box_value(f.address_of(), boxed_ty))
+            }
```

Comment:
Can you update the function-level comment regarding this case?

Reply:
Done!